llvm/lib/CodeGen/GlobalISel/CallLowering.cpp

```diff
@@ -121,6 +132,8 @@ bool CallLowering::lowerCall(MachineIRBuilder &MIRBuilder, const CallBase &CB,
     CanBeTailCalled = false;
   }
 
+  if (!hasConvergenceEntryToken(CB))
```

Comment:
Should this go under a CanBeTailCalled check?

Reply:
Don't understand this part; if there's any relation between tail call and convergence I would expect it to be the opposite

Reply:
Ping? Why is this here and/or backwards?

Reply:
It's just plain wrong. I no longer remember what I was doing when I put it in, but I have now implemented support for tokens on tail calls. Will push it along with responses to the other comments.